clang/include/clang/Format/Format.h

```diff
@@ -3353,7 +3353,12 @@ struct FormatStyle {
   }
   bool isTableGen() const { return Language == LK_TableGen; }
 
-  /// Language, this format style is targeted at.
+  /// The language that this format style targets.
+  /// \note
+  ///  You can also specify the language (``Cpp`` or ``ObjC``) for ``.h`` files
+  ///  by adding a ``// clang-format Language:`` line before the first
+  ///  non-comment and non-empty line, e.g. ``// clang-format Language: ObjC``.
```

Comment:
Should we allow an empty line? I for one have always the doxygen comment at the top and would use a blank line between other comments.

Reply:
Yes, and that’s what it means because empty lines are non-comment lines too.

Reply:
Before first non-comment line:
```cpp
/// \version 20
// clang-format Language: ObjC

#ifndef FOO_H
```
Before first non-empty line:
```cpp
 
// clang-format Language: ObjC
/// \version 20
#ifndef FOO_H
```
Before first non-comment (and non-empty) line:
```cpp
/// \version 20

// clang-format Language: ObjC

#ifndef FOO_H
```
The latter basically means "before the first non-comment token".